js/views/internal_links_support.js

```diff
@@ -179,40 +179,172 @@ var InternalLinksSupport = function(reader) {
     this.processLinkElements = function($iframe, spineItem) {
 
-        var epubContentDocument = $iframe[0].contentDocument;
-
-        $('a', epubContentDocument).click(function (clickEvent) {
-            // Check for both href and xlink:href attribute and get value
-            var href;
-            if (clickEvent.currentTarget.attributes["xlink:href"]) {
+        var settings = reader.viewerSettings();
+        if (!settings || typeof settings.epubPopupFootnotes === "undefined")
+        {
+            //defaults
+            settings = new ViewerSettings({});
+        }
+        var epubPopupFootNotesAreEnabled = settings.epubPopupFootnotes;
+        if (epubPopupFootNotesAreEnabled) {
+            var doc = ( $iframe[0].contentWindow || $iframe[0].contentDocument ).document;
+            var $elements = $("*[epub\\:type]", doc);
+            if ($elements.length) {
                 
-                href = clickEvent.currentTarget.attributes["xlink:href"].value;
-            }
-            else {
-                href = clickEvent.currentTarget.attributes["href"].value;
+                var sources = [];
+                var targets = [];
+                
+                $elements.each(function(i) {
+                    
+                    if ($elements[i].localName !== "a"
+                        && $elements[i].localName !== "aside") {
+                        return true; // continue
+                    }
+                    
+                    var $element = $($elements[i]);
+                    var epubType = $element.attr("epub:type"); 
+
+                    if (epubType == "noteref" && $element[0].localName === "a") {
+                        sources.push($element);
+                    } else if ((epubType == "footnote" || epubType == "note" || epubType == "rearnote") && $element[0].localName === "aside") {
+                        targets.push($element);
+                    }
+                });
+                
+                for (var i = 0; i < targets.length; i++) {
+                    
+                    var target = targets[i]; 
+                    
+                    var targetID = target.attr("id");
+                    
+                    if (!targetID) {
+                        console.warn("Aside footnote has no ID!");
+                        continue;
+                    }
+                        
+                    for (var j = 0; j < sources.length; j++) {
+                        var source = sources[j];
+                        
+                        var sourceIDREF = source.attr("href");
+                            
+                        if (!sourceIDREF) {
+                            console.warn("Link has no HREF!");
+                            continue;
+                        }
+                        
+                        if (sourceIDREF === ("#" + targetID)) {
+                            
+                            target.hide();
+                            
```

Comment:
Question: do we preserve the visibility of footnote content (DOM fragment), so that CFI bookmarks / annotations work as usual? I think yes. To be discussed.